go-controller/pkg/allocator/pod/pod_annotation.go

```diff
@@ -643,3 +644,27 @@ func AddRoutesGatewayIP(
 
 	return nil
 }
+
+func shouldSkipAllocateIPsError(err error, networkAllocated bool, ipamClaim *ipamclaimsapi.IPAMClaim) bool {
```

Comment:
nit: doc string explaining logic of the function (no need to repush if this is the only comment)